pandas/core/frame.py

```diff
@@ -3311,7 +3329,7 @@ def _box_col_values(self, values, loc: int) -> Series:
     # ----------------------------------------------------------------------
     # Unsorted
 
-    def query(self, expr, inplace=False, **kwargs):
+    def query(self, expr: str, inplace: bool = False, **kwargs) -> Optional[DataFrame]:
```

Comment:
i don't think this is correct for the output type

Reply:
I think this is right. From docstring:
```
DataFrame resulting from the provided query expression or None if ``inplace=True``.
```

Reply:
no my point is that the doc-string is wrong. this can be a Series or Scalar. basically this can be anything. I would remove it for now.

Reply:
Ok - removed (and will fix the docstring in follow-on, too)